### PR TITLE
feat(files): Allow to configure Windows filename compatibility in the settings

### PR DESCRIPTION
**Pull request by susnux**
* For #44963 

## Summary

This adds an admin setting to toggle Windows filename compatibility.

image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/lib/Service/SettingsService.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			$basenames = array_unique(array_values(array_diff($this->filenameValidator->getForbiddenBasenames(), self::WINDOWS_BASENAMES)));
+			$characters = array_unique(array_values(array_diff($this->filenameValidator->getForbiddenCharacters(), self::WINDOWS_CHARACTERS)));
+			$extensions = array_unique(array_values(array_diff($this->filenameValidator->getForbiddenExtensions(), self::WINDOWS_EXTENSION)));
+		}
+		$values = [
+			'forbidden_filename_basenames' => empty($basenames) ? null : $basenames,
+			'forbidden_filename_characters' => empty($characters) ? null : $characters,
+			'forbidden_filename_extensions' => empty($extensions) ? null : $extensions,
+		];
+		$this->logger->warning('Setting values', $values);
```
Comment (Altahrim) [inline apps/files/lib/Service/SettingsService.php:61]: No need to be a warning anymore?

**Review by Altahrim — APPROVED** [review]

File: apps/files/lib/Service/SettingsService.php (hunk 1 above)
Comment (susnux) [reply apps/files/lib/Service/SettingsService.php:61]: autsch. printf debugging :see_no_evil: 

```suggestion
```

**Review by artonge — APPROVED** [review]

**Comment by ThibautPlg** [comment]
Hi,
Very interesting and needed feature, thanks for the dev. 
Is it possible to trigger this using occ (`config:system:set` or something)?  
Best regards,

**Comment by susnux** [comment]
@ThibautPlg currently not you would have to set every setting manually so manually configure invalid characters and extensions. But this is a planed feature

**Pull request merged**